tests/python/test_runtime.py

```diff
@@ -1,6 +1,82 @@
 import taichi as ti
 from taichi import make_temp_file
-import sys, os
+import sys, os, copy
+from contextlib import contextmanager
+import pytest
+
+
+@contextmanager
+def patch_os_environ_helper(custom_environ: dict):
+    """
+    Temporarily patch os.environ for testing.
+    Originally created by @rexwangcc in test_cli.py
+    @archibate tweaked this method to be an os.environ patcher.
+    """
+    custom_environ = copy.deepcopy(custom_environ)
+    for key in os.environ.keys():
+        custom_environ[key] = os.environ[key]
+    try:
+        cached_environ = os.environ
+        os.environ = custom_environ
+        yield os.environ
+    finally:
+        os.environ = cached_environ
```

Comment:
@rexwangcc's method in https://github.com/taichi-dev/taichi/pull/1391#discussion_r449737022, not sure if I got the point in adapt to `os.environ`. (Maybe we could move this to `misc/test.py` later?)

Reply:
I think this monkey patch is being used appropriately here, this will yield the patched `os.environ` and return the call stack as if it reaches `__enter__` at line 21 and come back to clean up once it reaches `__exit__`, I only have 2 minor comments:

1. Depends on what environment variable you are trying to patch, 
```python
for key in os.environ.keys():
        custom_environ[key] = os.environ[key]
```
after the `custom_environ = copy.deepcopy(custom_environ)` might override the `custom_environ ` with existing old values. Imagine:
```diff
os.environ => {"good": "keep", "bad": "please patch me"}

after patch with ({"bad": "異議あり!", "ugly": "I'm new here!"})

+ you want: os.environ => {"good": "keep", "bad": "異議あり!", "ugly": "I'm new here!"}
- actual: os.environ => {"good": "keep", "bad": "please patch me", "ugly": "I'm new here!"}

```
which may not be what you want (ignore this if overriding the keys is what's expected here). A simple key presence check within `for` might solve this. 

2. It might be more straightforward to keep the testing utilities close to the test cases, i.e. `tests/`, even if they are not being tested.

Reply:
Done! Now I also added a `excluded` to exclude `TI_*` configs, so that `TI_USE_UNIFIED_MEMORY=0 ti test runtime` also works.